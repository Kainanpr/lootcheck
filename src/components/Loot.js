/* eslint-disable react/prefer-stateless-function */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { fetchBitcoin } from '../redux/actions/bitcoin';

export class Loot extends Component {
  static propTypes = {
    fetchBitcoin: PropTypes.func,
  };

  static defaultProps = {
    fetchBitcoin: () => {},
  };

  componentDidMount() {
    const { fetchBitcoin } = this.props;

    fetchBitcoin();
  }

  render() {
    return (
      <h3>Bitcoin balance:</h3>
    );
  }
}

const mapStateToProps = state => ({
  balance: state.balance,
  bitcoin: state.bitcoin,
});

const mapDispatchToProps = {
  fetchBitcoin,
};

export default connect(mapStateToProps, mapDispatchToProps)(Loot);
