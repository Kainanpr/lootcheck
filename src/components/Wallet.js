import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { makeDeposit } from '../redux/actions/balance';

export class Wallet extends Component {
  state = {
    inputValue: undefined,
  }

  static propTypes = {
    balance: PropTypes.number,
    makeDeposit: PropTypes.func,
  };

  static defaultProps = {
    balance: undefined,
    makeDeposit: () => {},
  };

  changeInputValue = (event) => {
    this.setState({ inputValue: Number(event.target.value) });
  }

  makeDeposit = () => {
    const { inputValue } = this.state;
    const { makeDeposit } = this.props;

    makeDeposit(inputValue);
  }

  render() {
    const { balance } = this.props;
    const { inputValue } = this.state;

    return (
      <div>
        <h3 className="balance">Wallet balance: {balance}</h3>
        <br />
        <input
          className="input-wallet"
          onChange={this.changeInputValue}
          value={inputValue}
        />
        <button
          type="button"
          className="btn-deposit"
          onClick={this.makeDeposit}
        >
          Deposit
        </button>
      </div>
    );
  }
}

const mapStateToProps = state => (
  {
    balance: state,
  }
);

const mapDispatchToProps = {
  makeDeposit,
};

export default connect(mapStateToProps, mapDispatchToProps)(Wallet);
