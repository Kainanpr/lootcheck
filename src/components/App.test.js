import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Wallet from './Wallet';
import Loot from './Loot';
import App from './App';

configure({ adapter: new Adapter() });

describe('App', () => {
  const wrapper = shallow(<App />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('contains a connected Wallet component', () => {
    expect(wrapper.contains(<Wallet />)).toBe(true);
  });

  it('contains a connected Loot component', () => {
    expect(wrapper.contains(<Loot />)).toBe(true);
  });

  it('contains a link to the coindesk price page', () => {
    expect(wrapper.find('a').props().href).toBe('https://www.coindesk.com/price');
  });
});
