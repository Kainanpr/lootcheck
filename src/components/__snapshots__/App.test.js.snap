// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders properly 1`] = `
<div>
  <h2>
    Loot Check
  </h2>
  <hr />
  <ConnectFunction />
  <hr />
  <ConnectFunction />
  <div>
    Powered by 
    <a
      href="https://www.coindesk.com/price"
      target="_blank"
    >
      Coindesk
    </a>
  </div>
</div>
`;
